Api Gateway Dual GW Arch model documentation update (#2500)
hashicorp/web-unified-docs: content/consul/v2.0.x/content/docs/error-messages/api-gateway.mdx

Reviewer: This should not be present here.

Reviewer 2: This is for the readability of the content writer and is not affecting the user doc. So, lets keep it.

Reviewer: Remove repetition and clarify you are referring to the "External CRDs". Or are you referring the the [NonStandardCRDs](https://developer.hashicorp.com/consul/docs/reference/k8s/helm#v-connectinject-apigateway-managenonstandardcrds)?

```suggestion
Install the required CRDs. Refer to the `connectInject. apiGateway.manageExternalCRDs` stanza in [Consul API Gateway installation instructions](/consul/docs/north-south/api-gateway/k8s/enable).
```

Reviewer 2: Taken the suggestion as it is.

Reviewer: There is no benefit of putting this text into a JSON block. Keep consistency and style it like the previous error message:

```suggestion
Operation cannot be fulfilled on gatewayclassconfigs.consul.hashicorp.com "consul-api-gateway": the object has been modified; please apply your changes to the latest version and try again
```

Reviewer 2: changed the format of Error to : 
```log
Error: Operation cannot be fulfilled on gatewayclassconfigs.consul.hashicorp.com. The object "consul-api-gateway" has been modified. Please apply your changes to the latest version and try again.
```

Reviewer: When does a gateway controller attempt to update an object? Does it happen randomly, when running a Helm upgrade, or on a scheduled basis?

The second sentence looks out of place and better moved into the next two sections - it is literally the Impact and how it will Resolve itself.

Reviewer 2: This content has been carried on since the previous releases. 
Removed the second part of the statement .

Reviewer: Add here what you mentioned earlier. Mention that the resolution is automatic, and also elaborate that it will resolve itself by retrying what exactly?

Reviewer 2: Corrected the  mentioned resolution

## Helm installation failed: "no matches for kind"

```log
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: [unable to recognize "": no matches for kind "GatewayClass" in version "gateway.networking.k8s.io/v1" or "consul.hashicorp.com/v1beta1", unable to recognize "": no matches for kind "GatewayClassConfig" in version "api-gateway.consul.hashicorp.com/v1alpha1"]
```
**Conditions:**

> Consul API gateway generates this error when the required CRD files have not been installed in Kubernetes prior to installing Consul API gateway.

Reviewer 3: ```suggestion
Consul API gateway generates this error when you did not install the required CRD files in Kubernetes prior to installing Consul API gateway.
```

**Impact:**
The installation process typically fails after this error message is generated.

**Resolution:**
Install the required CRDs. Refer to the `connectInject. apiGateway.manageExternalCRDs` stanza in [Consul API gateway installation instructions](/consul/docs/north-south/api-gateway/k8s/enable).

## Operation cannot be fulfilled, the object has been modified

```log
Error: Operation cannot be fulfilled on gatewayclassconfigs.consul.hashicorp.com. The object "consul-api-gateway" has been modified. Please apply your changes to the latest version and try again.
```
**Conditions:**
This error occurs when the gateway controller attempts to update an object that has been modified previously.

**Impact:**
Excessive error logs are produced, but there is no impact to the functionality of the controller.

**Resolution:**

> On updating the object manually or re-applying the manifest shall solve the error.

Reviewer 4: ```suggestion
Update the object manually or re-apply the manifest. This shall solve the error.
```
